admin_guide/tcp_ingress_external_ports.adoc

```diff
@@ -0,0 +1,99 @@
+{product-author}
+{product-version}
+:data-uri:
+:icons:
+:experimental:
+:toc: macro
+:toc-title:
+
+toc::[]
+
+== Overview
+
+Cluster administrators can assign a unique IP address to a service that, if
+routed correctly, enables external traffic to reach the ports of that service.
+This external IP assignment allows you to expose an external port, enabling
+ingress traffic (TCP/UDP) routing to a definable backend. This is much simpler
+than trying to manage the port space of a limited number of shared IP addresses.
+
+There is support for both automatic and manual assignment of IP addresses, each
+address is guaranteed to be assigned to a limit of not more than one service.
+This ensures that each service can simply direct traffic from its chosen ports,
+regardless of the ports selected by other services.
+
+[[unique-external-ips-ingress-traffic-restrictions]]
+== Restrictions
+
+To use `*IngressIP*`, you can:
+
+- Select an IP address with the existing `*ExternalIP*` ingress.
+- Have an IP address assigned from a pool. In this case, {product-title} implements a bare-metal version of the LoadBalancer service type and assigns IP addresses to the services.
++
+[CAUTION]
+====
+You must ensure that the IP address pool you assign terminates at a node in your
+cluster. You can use the existing `*oadm ipfailover*` to configure IP failover
+for the external IPs.
+====
+
+For manually-configured external IPs, potential port clashes are handled on a
+first-come, first-served basis. If you request a port, it is only available if
+it has not yet been assigned for that IP address. For example,
+`MongoDB service A` requests port 27017, and then `MongoDB service B` requests
+the same port; the first request gets the port, and the other receives a random
+port. This is similar to the route name allocation mechanism. However, port
+clashes are not an issue for ports assigned by the ingress controller, because
+each service has its own IP.
+
+[NOTE]
+====
+You can only assign external IPs to services in bare-metal environments. Ingress
+IPs can only be assigned if the cluster is not running in the cloud. In cloud
+environments, LoadBalancer-type services configure cloud-specific load
+balancers.
+====
+
+[[unique-external-ips-ingress-traffic-configure-cluster]]
+== Configuring the Cluster to Use Unique External IPs
+
+In non-cloud clusters, `ingressIPNetworkCIDR` is set by default to `172.46.0.0/16`. You could use the default if your cluster environment is not already using this private range. However, if you want to use a different range, then before you assign an ingress IP you must set xref:../install_config/master_node_configuration.adoc#master-node-config-network-config[`ingressIPNetworkCIDR`]
+in the *_/etc/origin/master-config.yaml_* file, then restart the master service.
+
+[[unique-external-ips-ingress-traffic-configure-service]]
+== Configuring an Ingress IP for a Service
+
+To assign an ingress IP:
+
+. Create a LoadBalancer service on your pod.
```

Comment:
Can we get and example service yaml? In particular we should show the `LoadBalancer` type.
```
apiVersion: v1
kind: Service
metadata:
  name: egress-1
spec:
  ports:
  - name: db
    port: 3306
  type: LoadBalancer
  selector:
    name: my-db-selector
```